bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/discovery/EpsonProjectorDiscoveryService.java

```diff
@@ -120,17 +129,20 @@ private synchronized void discover() {
 
                 if (thingProperties != null) {
                     // The MulticastListener found a projector, add it as new thing
-                    String uid = (String) thingProperties.get(THING_PROPERTY_MAC);
+                    String uid = (String) thingProperties.get(Thing.PROPERTY_MAC_ADDRESS);
                     String ipAddress = (String) thingProperties.get(THING_PROPERTY_HOST);
 
                     if (uid != null) {
                         logger.trace("Projector with UID {} discovered at IP: {}", uid, ipAddress);
 
                         ThingUID thingUid = new ThingUID(THING_TYPE_PROJECTOR_TCP, uid);
                         logger.trace("Creating epson projector discovery result for: {}, IP={}", uid, ipAddress);
-                        thingDiscovered(DiscoveryResultBuilder.create(thingUid).withProperties(thingProperties)
-                                .withLabel("Epson Projector " + uid).withRepresentationProperty(THING_PROPERTY_MAC)
-                                .build());
+                        thingDiscovered(
+                                DiscoveryResultBuilder.create(thingUid).withProperties(thingProperties)
+                                        .withLabel(translationProvider.getText(bundle,
+                                                "thing-type.epsonprojector.discovery.label", "Epson Projector",
+                                                localeProvider.getLocale()) + " " + uid)
```

Comment:
I would propose to use a placeholder for the `uid`, so the position is flexible for translations:
```
thing-type.epsonprojector.discovery.label = Epson Projector {0}
```

Reply:
fixed